src/main/java/de/bayern/gdi/gui/WMSMapSwing.java

```diff
@@ -765,6 +770,52 @@ public String getClickedPolygonID() {
         return this.selectedPolygonID;
     }
 
+
+    /** Button Group with Action Listener. */
+    @SuppressWarnings("serial")
+    private class ActionButtonGroup extends ButtonGroup {
+        private ActionListener btnGrpListener = new BtnGrpListener();
+        private EventListenerList listenerList = new EventListenerList();
+
+        @Override
+        public void add(AbstractButton b) {
+            b.addActionListener(btnGrpListener);
+            super.add(b);
+        }
+
+        public void addActionListener(ActionListener listener) {
+            listenerList.add(ActionListener.class, listener);
+        }
+
+        public void removeActionListener(ActionListener listener) {
+            listenerList.remove(ActionListener.class, listener);
+        }
+
+        protected void fireActionListeners() {
+            Object[] listeners = listenerList.getListenerList();
+            String actionCommand = "";
+            ButtonModel model = getSelection();
+            if (model != null) {
+                actionCommand = model.getActionCommand();
+            }
+            ActionEvent ae = new ActionEvent(this,
+                    ActionEvent.ACTION_PERFORMED,
+                    actionCommand);
+            for (int i = listeners.length - 2; i >= 0; i -= 2) {
+                if (listeners[i] == ActionListener.class) {
+                    ((ActionListener)listeners[i + 1]).actionPerformed(ae);
+                }
```

Comment:
http://stackoverflow.com/questions/6883604/listening-on-buttongroup-for-child-changes-and-print-selected-jradiobuttons